src/core/data.jl

```diff
@@ -96,36 +96,36 @@ function replicate_restoration_network(sn_data::Dict{String,<:Any}, count::Int,
         mn_data["nw"]["$n"] = deepcopy(sn_data_tmp)
     end
 
-    for n in 0:count-1
-        if n != count
-            mn_data["nw"]["$n"]["time_elapsed"] = get(mn_data["nw"]["$n"],"time_elapsed",1)*calc_repair_time_elapsed(sn_data, n, count)
-        end
-    end
-
-    return mn_data
-end
-
-
-""
-function calc_repair_time_elapsed(data::Dict{String,<:Any}, nw::Int, count::Int)
-    
     total_repairs = 0
     for j in ["gen","branch","storage"]
-        for (i,item) in data[j]
+        for (i,item) in sn_data[j]
             total_repairs = total_repairs + (get(item,"damaged",0)==1 ? 1 : 0)
         end
     end
 
     repairs_per_period = ceil(Int, total_repairs/count)
 
-    repairs_per_period*(nw+1)
-    if repairs_per_period*(nw+1) < total_repairs 
-        repairs = repairs_per_period
-    else
-        repairs = total_repairs - repairs_per_period*(nw)
-    end
+    mn_data["nw"]["0"]["repairs"] = 0
+    mn_data["nw"]["0"]["time_elapsed"] = repairs_per_period
+    mn_data["nw"]["0"]["repaired_total"] = 0
 
-    return repairs
+    for n in 1:count
+        if repairs_per_period*(n) < total_repairs 
+            mn_data["nw"]["$n"]["repairs"] = repairs_per_period
+        else
+            mn_data["nw"]["$n"]["repairs"] = total_repairs - mn_data["nw"]["$(n-1)"]["repaired_total"]
+        end
+
```

Comment:
if you first do,
```
time_elapsed = get(mn_data["nw"]["$n"], "time_elapsed", 1.0)
```
Then you can avoid the need for `if haskey(...)`, removing the code replication in the two branches also reduces the chance of a bug happening in only one of the branches.